.github/workflows/release.yml

```diff
@@ -0,0 +1,41 @@
+name: Release workflow
+
+on:
+  workflow_dispatch:
+
+env:
+  RUST_VERSION: rust:1.70-buster
+
+jobs:
+
+  build-and-push:
+    runs-on: ubuntu-latest
+    environment: release-recover-id-object
+    steps:
+    - name: Check out the repo
+      uses: actions/checkout@v4
+      with:
+        submodules: recursive
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+    - name: Log in to DockerHub
+      uses: docker/login-action@v3
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_TOKEN }}
+    - name: Extract Cargo package version
+      id: cargo_version
+      run: |
+        cd recover-id-object
+        VERSION=$(cargo metadata --no-deps --format-version 1 | jq -r '.packages[0].version')
+        echo "Extracted version is $VERSION"
+        echo "::set-output name=version::$VERSION"
+    - name: Build and push Docker image
+      uses: docker/build-push-action@v5
+      with:
+        context: .
+        file: recover-id-object/scripts/Dockerfile
+        push: true
+        build-args: |
+          build_image=${{ env.RUST_VERSION }}
+        tags: concordium/recover-id-object:${{ steps.cargo_version.outputs.version }}
```

Comment:
I could also use `concordium/recover-id-object:${{ steps.cargo_version.outputs.version }}-${{ github.run_attempt }}`

https://docs.github.com/en/actions/learn-github-actions/contexts

Reply:
Let's not.